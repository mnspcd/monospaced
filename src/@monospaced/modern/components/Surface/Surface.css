.mdrn .Surface--backgroundDark {
  color: var(--color-white);
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.mdrn .Surface--backgroundDark a {
  color: var(--color-blue-5);
}

.mdrn .Surface--backgroundDark a:hover {
  color: var(--color-blue-4);
}

.mdrn .Surface--backgroundDark a:active {
  color: var(--color-blue-3);
}
