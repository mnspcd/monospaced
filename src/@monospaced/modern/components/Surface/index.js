export { default, SurfaceContext } from "./Surface";
