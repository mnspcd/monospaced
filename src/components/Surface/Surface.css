.Surface--backgroundDark {
  color: var(--mnspcd-color-white);
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.Surface--backgroundDark a {
  color: var(--mnspcd-color-blue-5);
}

.Surface--backgroundDark a:hover {
  color: var(--mnspcd-color-blue-4);
}

.Surface--backgroundDark a:active {
  color: var(--mnspcd-color-blue-3);
}

.Surface--backgroundLightBlue {
  color: var(--mnspcd-color-blue-9);
}

.Surface--graphPaper {
  background-image:
    linear-gradient(to right, var(--mnspcd-color-blue-2) calc(1rem / 16),
    transparent calc(1rem / 16)),
    linear-gradient(to bottom, var(--mnspcd-color-blue-2) calc(1rem / 16),
    transparent calc(1rem / 16));
  background-size: 1.5em 1.5em;
}

.Surface--graphPaper.Surface--backgroundDark {
  background-image:
    linear-gradient(to right, var(--mnspcd-color-blue-8) calc(1rem / 16),
    transparent calc(1rem / 16)),
    linear-gradient(to bottom, var(--mnspcd-color-blue-8) calc(1rem / 16),
    transparent calc(1rem / 16));
}
