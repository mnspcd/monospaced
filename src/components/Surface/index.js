export { default, SurfaceConsumer } from "./Surface";
