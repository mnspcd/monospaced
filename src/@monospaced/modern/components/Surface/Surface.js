import classNames from "classnames";
import PropTypes from "prop-types";
import React from "react";

import tokens from "../../tokens";

import "./Surface.css";

const backgroundColors = {
  white: tokens["color-white"],
  "grey-light": tokens["color-grey-1"],
  "grey-dark": tokens["color-grey-9"],
  black: tokens["color-black"],
};

const { Consumer, Provider } = React.createContext();

/**
 * Use `Surface` to display components on different background colors.
 */
const Surface = ({ backgroundColor = "white", children }) => {
  const surfaceBackground =
    backgroundColor.endsWith("dark") || backgroundColor.endsWith("black")
      ? "dark"
      : backgroundColor.replace("color-", "");

  return (
    <Provider value={surfaceBackground}>
      <div
        className={classNames({
          Surface: true,
          [`Surface--backgroundDark`]: surfaceBackground === "dark",
        })}
        style={{
          backgroundColor: backgroundColors[backgroundColor],
        }}
      >
        {children}
      </div>
    </Provider>
  );
};

const colorKeys = Object.keys(backgroundColors);

Surface.backgroundColors = colorKeys;

Surface.propTypes = {
  /**
   * Surface color variant.
   */
  backgroundColor: PropTypes.oneOf(colorKeys),

  /**
   * Content to display on the Surface.
   */
  children: PropTypes.node.isRequired,
};

export default Surface;
export { Consumer as SurfaceConsumer };
