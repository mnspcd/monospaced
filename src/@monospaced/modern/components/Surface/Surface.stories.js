import { select } from "@storybook/addon-knobs";
import React from "react";

import Surface from "./Surface";

export default {
  component: Surface,
  title: "Components/Surface",
};

export const basic = () => (
  <Surface backgroundColor="grey-light">
    <div className="u-monospaced" style={{ padding: "1em" }}>
      Text <br />
      <a href="#">Link</a>
    </div>
  </Surface>
);

export const white = () => (
  <Surface backgroundColor="white">
    <div className="u-monospaced">
      Text <br />
      <a href="#">Link</a>
    </div>
  </Surface>
);

export const greyLight = () => (
  <Surface backgroundColor="grey-light">
    <div className="u-monospaced" style={{ padding: "1em" }}>
      Text <br />
      <a href="#">Link</a>
    </div>
  </Surface>
);

export const greyDark = () => (
  <Surface backgroundColor="grey-dark">
    <div className="u-monospaced" style={{ padding: "1em" }}>
      Text <br />
      <a href="#">Link</a>
    </div>
  </Surface>
);

export const black = () => (
  <Surface backgroundColor="black">
    <div className="u-monospaced" style={{ padding: "1em" }}>
      Text <br />
      <a href="#">Link</a>
    </div>
  </Surface>
);

export const knobs = () => (
  <Surface
    backgroundColor={select(
      "backgroundColor",
      Surface.backgroundColors,
      "white",
    )}
  >
    <div className="u-monospaced">
      Text <br />
      <a href="#">Link</a>
    </div>
  </Surface>
);

knobs.story = { parameters: { docs: { disable: true } } };
