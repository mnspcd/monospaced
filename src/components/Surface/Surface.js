import PropTypes from "prop-types";
import React from "react";
import classNames from "classnames";

import "./Surface.css";
import { backgroundColors } from "../../system/color";

const { Consumer, Provider } = React.createContext();

const Surface = ({ backgroundColor, children }) => {
  const surfaceBackground =
    backgroundColor.endsWith("dark") || backgroundColor === "black"
      ? "dark"
      : backgroundColor.replace("mnspcd-color-", "");

  return (
    <Provider value={surfaceBackground}>
      <div
        className={classNames({
          Surface: true,
          [`Surface--backgroundDark`]: surfaceBackground === "dark",
        })}
        style={{
          backgroundColor: backgroundColors[backgroundColor],
        }}
      >
        {children}
      </div>
    </Provider>
  );
};

Surface.backgroundColors = Object.keys(backgroundColors);

Surface.defaultProps = { backgroundColor: "white" };

Surface.propTypes = {
  backgroundColor: PropTypes.oneOf(Surface.backgroundColors),
  children: PropTypes.node.isRequired,
};

export default Surface;
export { Consumer as SurfaceConsumer };
