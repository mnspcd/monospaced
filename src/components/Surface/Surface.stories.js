import React from "react";
import { select } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";

import Surface from "./Surface";

import typography from "../../system/typography";

storiesOf("Components/Surface", module).add("Component", () => {
  return (
    <div style={{ fontFamily: `${typography["font-monospaced"]}` }}>
      <Surface
        backgroundColor={select(
          "backgroundColor",
          Surface.backgroundColors,
          "white",
        )}
      >
        Text color <br />
        <a href="#">Link color</a>
      </Surface>
    </div>
  );
});
