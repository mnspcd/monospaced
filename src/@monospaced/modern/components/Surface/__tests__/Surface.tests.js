/* eslint-env jest */
import { render } from "@testing-library/react";
import React from "react";

import Surface from "..";

describe("Surface component", () => {
  it("should render correctly", () => {
    const { container } = render(<Surface>children</Surface>);

    expect(container).toMatchSnapshot();
  });

  it("should render dark background correctly", () => {
    const { container } = render(
      <Surface backgroundColor="grey-dark">children</Surface>,
    );

    expect(container).toMatchSnapshot();
  });
});
