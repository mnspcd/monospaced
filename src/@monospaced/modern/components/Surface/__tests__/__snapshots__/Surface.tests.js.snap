// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Surface component should render correctly 1`] = `
<div>
  <div
    class="Surface"
    style="background-color: rgb(255, 255, 255);"
  >
    children
  </div>
</div>
`;

exports[`Surface component should render dark background correctly 1`] = `
<div>
  <div
    class="Surface Surface--backgroundDark"
    style="background-color: rgb(22, 22, 22);"
  >
    children
  </div>
</div>
`;
